File: dotnet/src/Jobing.Domain/Entities/Filter.cs
namespace Jobing.Domain.Entities;

public class Filter
{
    public int Id { get; set; }
    public Dictionary<string, string> Name { get; set; } = new();
    public string Key { get; set; } = string.Empty;
    public int SortOrder { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    // Navigation
    public List<FilterOption> Options { get; set; } = new();
}

File: src/Modules/Filters/Entities/Filter.js
import { DataTypes } from 'sequelize';
import sequelize from '../../../Config/Database.js';

const Filter = sequelize.define('Filter', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    key: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true
    },
    name: {
        type: DataTypes.JSONB,
        allowNull: false,
        defaultValue: {}
    },
    sortOrder: {
        type: DataTypes.INTEGER,
        defaultValue: 0
    },
    isActive: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    deletedAt: {
        type: DataTypes.DATE,
        allowNull: true
    }
}, {
    tableName: 'filters',
    timestamps: true,
    paranoid: false, // We'll handle soft deletes manually or switch this to true
    createdAt: 'createdAt',
    updatedAt: 'updatedAt'
});

export default Filter;
